Python/src/device.h

```diff
@@ -64,6 +66,10 @@ namespace pydml
         Microsoft::WRL::ComPtr<ID3D12CommandQueue> m_commandQueue;
         Microsoft::WRL::ComPtr<ID3D12CommandAllocator> m_commandAllocator;
         Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> m_commandList;
+        Microsoft::WRL::ComPtr<gpgmm::d3d12::ResourceAllocator> m_resourceAllocator;
+        
+        // Residency management is used to handle oversubscribing of video memory.
+        gpgmm::d3d12::ResidencyManager* m_residencyManager = nullptr;
```

Comment:
Can you confirm that the lifetime of the `ResidencyManager` is fully managed by the `ResourceAllocator`?

Reply:
It also looks like your fork is behind. I'm not able to build this change from your fork without compile errors.

Reply:
Thank you, @wchao1115.

> Can you confirm that the lifetime of the ResidencyManager is fully managed by the ResourceAllocator?

Yup, that's correct. Comment updated.

> It also looks like your fork is behind. 

Updated. Hopefully that fixes your issue.

Reply:
I still am not able to build with your fork. I think you'll need to pull the latest from master. 

> C:\DirectML-Samples\Python\src\module.cpp(387,11): error C2672: 'pybind11::module::def': no matching overloaded function found [C:\DirectML-Samples\Python\build\temp.win
-amd64-3.7\Debug\pydirectml.vcxproj]
C:\DirectML-Samples\Python\src\module.cpp(387,11): error C2783: 'pybind11::module &pybind11::module::def(const char *,Func &&,const Extra &...)': could not deduce templa
te argument for 'Func' [C:\DirectML-Samples\Python\build\temp.win-amd64-3.7\Debug\pydirectml.vcxproj]

Reply:
@wchao1115 Did DirectMLX go out of sync on your machine?

I had to delete `.\Python\dependencies`  AND run  `python setup.py clean` before `python setup.py install` (due to 17603b782b945c56fca1956bc3d6f73bf0b86d10). My folk, [bbernhar:gpgmm](https://github.com/bbernhar/DirectML-Samples/tree/gpgmm) is (already) on-top of `microsoft:master`.

Give that a try.

Reply:
Thanks. That builds now. I thought you already merged to the master branch in your fork. Looks good to me.